<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>705</width>
    <height>473</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="btnClose">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>60</y>
      <width>80</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>close</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnUi">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>190</y>
      <width>80</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>No1 UI</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnQObj">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>190</y>
      <width>80</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>No2 QObj</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnLambda">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>190</y>
      <width>80</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>No3 Lambda</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnPointer">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>190</y>
      <width>80</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>No4 Pointer</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnReadFile">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>260</y>
      <width>80</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>读取文件</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnWriteFile">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>260</y>
      <width>80</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>写入文件</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnStreamRead">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>260</y>
      <width>80</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Stream 读取</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnStreamWrite">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>260</y>
      <width>80</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Stream 写入</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnQFileDialogOpen">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>310</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>打开文件选择框</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btnSaveFile">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>310</y>
      <width>80</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>保存文件</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>705</width>
     <height>19</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>btnClose</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>319</x>
     <y>155</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>236</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
